x-pack/plugins/enterprise_search/public/applications/elasticsearch/components/layout/page_template.tsx

```diff
@@ -19,12 +21,14 @@ export const EnterpriseSearchElasticsearchPageTemplate: React.FC<PageTemplatePro
   pageViewTelemetry,
   ...pageTemplateProps
 }) => {
+  const { initialData } = useEnterpriseSearchData();
+
   return (
     <EnterpriseSearchPageTemplateWrapper
       {...pageTemplateProps}
       solutionNav={{
         name: ELASTICSEARCH_PLUGIN.NAME,
-        items: useEnterpriseSearchNav(),
+        items: useEnterpriseSearchNav(initialData.access || NO_ACCESS),
```

Comment:
Suggestion: we're doing this same thing in three different places now, which is probably going to lead to some missed implementations in the future. 

Why don't we create a new EnterpriseSearchPageTemplate (or maybe: EnterpriseSearchPageTemplateWrapper that implements this nav in one spot, and then extend the individual solution templates using it from there?

That template could pick up the real value either from Kea, or from the context created below

Reply:
I get the potential for confusion, but I don't think that's a massive concern. Essentially, every new page going forward should use the new TemplateWrapper we'd write here. The alternative navigations really only exist because App Search and Workplace Search are legacy apps we're deprecating soon, and we don't want to touch them. 

My ideal implementation would be renaming this wrapper to `LegacyEnterpriseSearchPageTemplateWrapper`, and writing a new `EnterpriseSearchPageTemplateWrapper` that has the EnterpriseSearchNav inserted. We use the latter in the places that aren't App Search and Workplace Search (so the stuff you're touching in this PR) and the former in App Search and Workplace Search.

And yeah, I noticed the deprecation as well. You can address that in this PR if you want, looks like you just need to point the import at a new location.

Reply:
to capture what we discussed at the end of the frontend alignment meeting:

If we can deliver this in 8.5 I'm fine with doing the refactor to introduce a new Template and clean-up for deprecated KibanaPageTemplate in this PR. If this needs to land in 8.4.0 then I'd want to keep this PR targeted and do a follow-up with the refactors.

Reply:
We've sidestepped this by just pushing the `useValues` for `KibanaLogic` down into `useEnterpriseSearchNav`.